src/main/haskell/kore/src/Kore/Builtin/KEqual.hs

```diff
@@ -139,25 +135,24 @@ evalKEq
         ( AttemptedAxiom Object variable
         , SimplificationProof Object
         )
-evalKEq true tools substitutionSimplifier _ (valid :< app) =
+evalKEq true _ _ _ (valid :< app) =
     case applicationChildren of
         [t1, t2] -> evalEq t1 t2
         _ -> Builtin.wrongArity (if true then eqKey else neqKey)
   where
     false = not true
     Valid { patternSort } = valid
     Application { applicationChildren } = app
-    evalEq t1 t2 = do
-        (result, _proof) <- makeEvaluate tools substitutionSimplifier ep1 ep2
-        case () of
-            _ | OrOfExpandedPattern.isTrue result ->
-                purePatternAxiomEvaluator (Bool.asInternal patternSort true)
-              | OrOfExpandedPattern.isFalse result ->
-                purePatternAxiomEvaluator (Bool.asInternal patternSort false)
-              | otherwise -> notApplicableAxiomEvaluator
-      where
-        ep1 = ExpandedPattern.fromPurePattern t1
-        ep2 = ExpandedPattern.fromPurePattern t2
+    evalEq t1 t2 = purePatternAxiomEvaluator $
+        mkOr
```

Comment:
Let's make sure this gets simplified, either by calling the pattern simplifier from the top, or by using the simplified-pattern constructors.